python/ray/util/iter.py

```diff
@@ -198,6 +198,37 @@ def for_each(self, fn: Callable[[T], U]) -> "ParallelIterator[U]":
         return self._with_transform(lambda local_it: local_it.for_each(fn),
                                     ".for_each()")
 
+    def for_each_concur(self,
+                        fn: Callable[[T], U],
+                        max_concur=2,
+                        resources=None) -> "ParallelIterator[U]":
+        """Remotely apply fn to each item in this iterator, at most max_concur at a
+        time.
+
```

Comment:
When should a user use this?

Reply:
I can add this to the documentation, but it's needed to achieve a higher degree of parallelism. Currently for_each operations are serialized within an actor, and actor overhead is too high to have a lot of parallelism

Reply:
(haha i meant, the answer to this question should go in the documentation)